bitseq/sequence.go

```diff
@@ -240,8 +251,11 @@ func (h *Handle) set(ordinal uint32, any bool, release bool) (uint32, error) {
 			bytePos, bitPos = ordinalToPos(ordinal)
 		} else {
 			if any {
-				bytePos, bitPos, err = getFirstAvailable(h.head)
+				bytePos, bitPos, err = getFirstAvailable(h.head, start)
 				ret = posToOrdinal(bytePos, bitPos)
+				if end != 0 && end < ret {
```

Comment:
I thought the inner function `GetFirstAvailable()` already checks for the upper limit `h.bits` and returned an error if we go over (this would happen if the number of bits is not a multiple of 32). But it doesn't...

I will make the change you suggested.
Thanks.